airflow/contrib/hooks/gcp_api_base_hook.py

```diff
@@ -86,6 +90,24 @@ def _authorize(self):
                                        'use a JSON key file.')
             else:
                 raise AirflowException('Unrecognised extension for key file.')
+        else:
+            if not scope:
+                raise AirflowException('Scope should be defined when using key JSON.')
+            scopes = [s.strip() for s in scope.split(',')]
+
+            # Get credentials from JSON data provided in the UI.
+            try:
+                keyfile_dict = json.loads(key_json)
+
+                # Depending on how the JSON was formatted, it may contain
+                # escaped newlines. Convert those to actual newlines.
+                keyfile_dict['private_key'] = keyfile_dict['private_key'].replace(
```

Comment:
curious in what scenario this conversion is effective,  as oppose to a json strng that was not formatted properly to begin with.

Reply:
I saw this when I loaded the JSON file into a Python REPL, then used `json.dumps()` to convert it to a string. Probably a mistake on my part, and this would not be necessary if the file were converted some other ways. I left it because I'm pretty sure it's harmless if not needed, and potentially very helpful. Does it seem reasonable? Should I update the comment to include this additional detail?